pages/unlock-publishing.tsx

```diff
@@ -0,0 +1,254 @@
+                        hover:text-white-default hover:bg-blue-light active:text-white-default active:bg-blue-active active:top-0.5
+                        focus:outline-[3px] focus:outline-blue-focus focus:outline focus:outline-offset-2 focus:bg-blue-focus focus:text-white-default disabled:cursor-not-allowed disabled:text-gray-500
+                      `}
+                    disabled={submitting}
+                  >
+                    {t("submitButton", { ns: "common" })}
+                  </Button>
+                  <StyledLink
+                    href={`/${i18n.language}/myforms/`}
+                    className={` 
+                          no-underline visited:text-black-default 
+                          bg-white-default text-black-default border-black-default py-4 px-8 rounded-lg border-2 border-solid
+                          hover:text-white-default hover:bg-blue-light active:text-white-default active:bg-blue-active active:top-0.5
+                          focus:outline-[3px] focus:outline-blue-focus focus:outline focus:outline-offset-2 focus:bg-blue-focus focus:text-white-default disabled:cursor-not-allowed disabled:text-gray-500
+                        `}
+                  >
+                    {t("unlockPublishing.skipStepButton")}
+                  </StyledLink>
+                </div>
+              </form>
+            </>
+          )}
+        </Formik>
+      )}
+      {hasRequestedPublishing && (
+        <>
+          <h1>{t("unlockPublishingSubmitted.title")}</h1>
+          <h2>{t("unlockPublishingSubmitted.whatNext.title")}</h2>
+          <p>{t("unlockPublishingSubmitted.whatNext.paragraph1")}</p>
+          <p className="mt-8 font-bold">{t("unlockPublishingSubmitted.whatNext.paragraph2")}</p>
+          <p>{t("unlockPublishingSubmitted.whatNext.paragraph3")}</p>
+          <p className="mt-8">{t("unlockPublishingSubmitted.whatNext.paragraph4")}</p>
+          <div className="flex mt-14">
+            <StyledLink
+              href={`/${i18n.language}/myforms/`}
+              className={` 
+                no-underline visited:text-white-default mr-8
+                bg-blue-dark text-white-default border-black-default py-4 px-8 rounded-lg border-2 border-solid
+                hover:text-white-default hover:bg-blue-light active:text-white-default active:bg-blue-active active:top-0.5
+                focus:outline-[3px] focus:outline-blue-focus focus:outline focus:outline-offset-2 focus:bg-blue-focus focus:text-white-default disabled:cursor-not-allowed disabled:text-gray-500
+              `}
+            >
+              {t("continue", { ns: "common" })}
+            </StyledLink>
+          </div>
+        </>
+      )}
+    </div>
+  );
+}
+
+export const getServerSideProps = requireAuthentication(async ({ user: { ability }, locale }) => {
+  {
+    checkPrivileges(ability, [{ action: "update", subject: "FormRecord", field: "isPublished" }]);
```

Comment:
@bryan-robitaille I think this is closer to being right but the "isPublished", I'm not sure that is how "field" should be used?

Reply:
The check is right but with this logic implentation a user that does not have the ability to publish would never be able to reach this page.  The `checkPrivileges` with throw an access control error and the user will be redirected to a Forbidden page if not caught.  There is another function in the `privileges` lib that we can slightly modify and use here: `checkPrivilegesAsBoolean`.  We can then redirect the user back to myforms if they have already have the publishing ability on their profile.